pkg/agent/route/route_linux.go

```diff
@@ -128,20 +134,39 @@ type Client struct {
 	egressRoutes sync.Map
 	// The latest calculated Service CIDRs can be got from serviceCIDRProvider.
 	serviceCIDRProvider servicecidr.Interface
+	// nodeNetworkPolicyIPSetsIPv4 caches all existing IPv4 ipsets for NodeNetworkPolicy.
+	nodeNetworkPolicyIPSetsIPv4 sync.Map
+	// nodeNetworkPolicyIPSetsIPv6 caches all existing IPv6 ipsets for NodeNetworkPolicy.
+	nodeNetworkPolicyIPSetsIPv6 sync.Map
+	// nodeNetworkPolicyIPSetsIPv4 caches all existing IPv4 iptables chains and rules for NodeNetworkPolicy.
+	nodeNetworkPolicyIPTablesIPv4 sync.Map
+	// nodeNetworkPolicyIPSetsIPv6 caches all existing IPv4 iptables chains and rules for NodeNetworkPolicy.
+	nodeNetworkPolicyIPTablesIPv6 sync.Map
+	// nodeNetworkPolicyIPTablesDeterministic represents whether to write iptables chains and rules for NodeNetworkPolicy
+	// deterministically when syncIPTables is called. Enabling it may carry a performance impact. It's disabled by default
+	// and should only be used in testing.
+	nodeNetworkPolicyIPTablesDeterministic bool
```

Comment:
perhaps just `deterministic` so it can be used in all cases. Not worth to have a bool for every case.